metadata/src/utils/validation.rs

```diff
@@ -75,6 +75,7 @@ fn get_field_hash(
     registry: &PortableRegistry,
     field: &Field<PortableForm>,
     cache: &mut HashMap<u32, CachedHash>,
+    precomputed: Option<&HashMap<u32, [u8; HASH_LEN]>>,
```

Comment:
I find it a bit hard to distinguish between "cache and precomputed" here, to me those seems very similar and wonder if it makes sense to merge them to single type instead?

It looks like `cache` is an empty hashmap in plenty of places and precomputed is an Option<HashMap>... would be nice unify that

Reply:
I agree that it's confusing, although I wouldn't merge it with the cache (see comment below :))

Reply:
Also a type-alias for hash would nice to avoid repeating `[u8; HASH_LEN]`